qiskit/circuit/quantumcircuit.py

```diff
@@ -1062,7 +1090,24 @@ def __init__(
             regs = tuple(int(reg) for reg in regs)  # cast to int
         self._base_name = None
         self.name: str
-        """A human-readable name for the circuit."""
+        """A human-readable name for the circuit.
+        
+        Example:
+
+            .. plot::
+                :include-source:
+                :nofigs:
+                :context: reset
+
+                from qiskit import QuantumCircuit
+
+                qc = QuantumCircuit(2, 2, name="my_circuit")
+                print(qc.name)
```

Comment:
Minor stylistic: is there any chance that including the `from qiskit import QuantumCircuit` line verbatim in many examples will be too noisy for readers of the documentation?